album-recommendation-java/README.md

```diff
@@ -1,72 +1,75 @@
 <!-- Copyright 2019 Oath Inc. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root. -->
-# Vespa sample applications - Basic stateless Vespa application
+# Vespa sample applications - album recommendations java
 
 Extends the [album-recommedations](../album-recommedations) sample application with a Searcher component in Java
-which does query and result processing.
+which does query and result processing. Refer to
+[developing searchers](http://docs.vespa.ai/documentation/searcher-development.html) for more information.
 
-Refer to
-[developing searchers](http://docs.vespa.ai/documentation/searcher-development.html)
-for more information.
+This sample app introduces how to write system tests and how to integrate with a CI/CD pipeline
 
 See [getting started](http://cloud.vespa.ai/getting-started.html) for troubleshooting.
 
+Security notes:
+*   To deploy to your instance, a _personal deploy key_ is required.
+    See step 4 below.
+*   To read and write to the instance's endpoint, a certificate and a private key is required.
+    See step 2 below.
+    Find more details in [Data Plane](https://cloud.vespa.ai/security-model.html#data-plane), see _Client certificate_.
+*   Below it is assumed names and key paths are put in `pom.xml`.
+    Alternatively, submit these values using  `mvn ... -DapiKeyFile=/path/to/key.pem`. 
+*   Instead of using a _personal deploy key_, one can also deploy using the console, see step 5 in
+    [../album-recommendation/README.md](album-recommendation).
+
 
-## Getting started
-Prerequisites: git, Java 11, mvn 3.6.1 and a X.509 certificate.
-The certificate is used to access the application's endpoints.
 
-1.  Go to http://console.vespa.ai/, click "Create application"
+## Getting started
+Prerequisites: git, Java 11, mvn 3.6.1 and certificates/keys (above).
```

Comment:
Make this requirement "deployment keys (above)" and then add "openssl" to the list of requirements since we make dataplane keys and certificates below?

Reply:
Aha. I see we ask for downloading the key in the list as well, then it's not a requirement? Just add openssl?